pkg/logs/input/container/scanner.go

```diff
@@ -226,6 +226,8 @@ func (s *Scanner) newDockerClient() (*client.Client, error) {
 	if err != nil {
 		return nil, err
 	}
+	// assure the client can communicate with the server
+	client.UpdateClientVersion("")
 	v, err := client.ServerVersion(context.Background())
```

Comment:
could we extract lines 230 and 231 into a method to ensure we always call `UpdateClientVersion("")` whenever we compute the server version?
also, could this benefit other parts of the agent? Would it be worth moving to common logic?

Reply:
FWI this is the logic infra+process agents use: https://github.com/DataDog/datadog-agent/blob/master/pkg/util/docker/docker_util.go#L106-L132

This fix is cherry-picked from our 1.12 compat fixes.

Reply:
I suggest keep using this piece of code and move to `docker_util` at some point, WDYT ?